src/SearchForm.cpp

```diff
@@ -248,7 +246,16 @@ void SearchResultList::MessageReceived(BMessage* message) {
 
 			int index = 0;
 			TiXmlElement* searchresult = document.FirstChildElement("searchresults");
-			for (TiXmlElement* e = searchresult->FirstChildElement("place"); e != NULL; e = e->NextSiblingElement("place")) {
+			// Also, we would want to tell the user when there are no matches.
+			TiXmlElement* e = searchresult->FirstChildElement("place");
+			if (e == NULL) {
+				SearchResultList_Data* itemData = new SearchResultList_Data();
+				AddItem(new BStringItem("No matches found!"), 0);
+				itemData->longitude = 0;
+				itemData->latitude = 0;
+				itemList.insert(std::pair<int, SearchResultList_Data*>(0, itemData));
```

Comment:
Would be good if you set the longitude and latitude to current location instead?